meta-balena-common/recipes-support/balena-config-vars/balena-config-vars/balena-config-vars

```diff
@@ -117,18 +111,34 @@ else
         if [ "$RESULT" = "4" ]; then
             echo "[ERROR] $0: Failed to parse '$CONFIG_PATH'."
             exit 1
-        else
-            if [ -n "${BALENA_CONFIG_VARS_CACHE}" ]; then
-                tmpfile=$(mktemp)
-                echo "${CONFIG_PARAMS}" | sed -e 's/^[[:space:]]*//' > "${tmpfile}"
-                chmod a+rx "${tmpfile}"
-                mv "${tmpfile}" "${BALENA_CONFIG_VARS_CACHE}"
-            fi
-            eval "$CONFIG_PARAMS"
         fi
+        test -n "${CONFIG_PARAMS}"
```

Comment:
Is this a leftover from testing?

Reply:
It sets the return code for the function @floion if `CONFIG_PARAMS` is empty it will return 1, and 0 otherwise.

Reply:
Right, I was thinking the script would be ended by the `exit 1` from above if parsing the config would fail.

Now come to think about it, wouldn't the function return with 0 when it goes through the else branch which follows this? Because that echo will return 0 itself and pass that code to the read_config calling function?

Reply:
hey @floion the function return code is the one from the last command. If it goes through the else branch:
```
        if [ "$RESULT" = "4" ]; then
            echo "[ERROR] $0: Failed to parse '$CONFIG_PATH'."
            exit 1
        else
            if [ -n "${BALENA_CONFIG_VARS_CACHE}" ]; then
                tmpfile=$(mktemp)
                echo "${CONFIG_PARAMS}" | sed -e 's/^[[:space:]]*//' > "${tmpfile}"
                chmod a+rx "${tmpfile}"
                mv "${tmpfile}" "${BALENA_CONFIG_VARS_CACHE}"
            fi
            eval "$CONFIG_PARAMS"
        fi
        test -n "${CONFIG_PARAMS}"
```
After the `mv` it will still run the `eval` and then `test` commands and return the exit code of the last one which is `test`.

Reply:
I can add a comment to the function to make it clear.